Traffic Ops TestGetMidServers unit test fails

<!--
************ STOP!! ************
If this issue identifies a security vulnerability, DO NOT submit it! Instead, contact
the Apache Software Foundation Security Team at security@trafficcontrol.apache.org and follow the
guidelines at https://www.apache.org/security/ regarding vulnerability disclosure.
-->

<!--
- For *SUPPORT QUESTIONS*, use the
[Traffic Control slack channels](https://traffic-control-cdn.slack.com) or [Traffic Control mailing lists](http://trafficcontrol.apache.org/mailing_lists/).
- Before submitting, please **SEARCH GITHUB** for a similar issue or PR. -->

## I'm submitting a ...
<!-- (check all that apply with "[x]") -->
<!--- security vulnerability (STOP!! - see above)-->
- [x] bug report
- [ ] new feature / enhancement request
- [ ] improvement request (usability, performance, tech debt, etc.)
- [ ] other <!--(Please do not submit support requests here - see above)-->

## Traffic Control components affected ...
<!-- (check all that apply with "[x]") -->
- [ ] CDN in a Box
- [ ] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [x] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:
<!-- Describe how the bug manifests / how the current features are insufficient. -->
About 12% of the time, [`TestGetMidServers`](https://github.com/apache/trafficcontrol/blob/e1d431b134/traffic_ops/traffic_ops_golang/server/servers_test.go#L269) fails with
```json
--- FAIL: TestGetMidServers (0.00s)
    servers_test.go:475: getMidServers expected: len(mid) == 1, actual: 0
FAIL
exit status 1
```

Tested at apache/trafficcontrol@e1d431b134

## Expected / new behavior:
<!-- Describe what the behavior would be without the bug / how the feature would improve Traffic Control -->
```json
PASS
ok      github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/server  0.006s
```

## <div id="testcase">Minimal reproduction of the problem with instructions:</div>
<!--
If the current behavior is a bug or you can illustrate your feature request better with an example,
please provide the *STEPS TO REPRODUCE* and include the applicable TC version.
-->
Run the test 100 times (takes under 1 second if precompiling the test)

```bash
go test -o servers_test ./traffic_ops/traffic_ops_golang/server;
failed=0;
for total in {1..100}; do
  if ! ./servers_test -test.run TestGetMidServers; then
    (( failed++ ));
  fi;
done;
printf 'Test failed %d times out of %d\n' $failed $total;
```

## Anything else:
<!-- e.g. stacktraces, related issues, suggestions how to fix -->
```go
PASS
ok  	github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/server	0.007s
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestGetMidServers (0.00s)
    servers_test.go:475: getMidServers expected: len(mid) == 1, actual: 0
FAIL
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestGetMidServers (0.00s)
    servers_test.go:475: getMidServers expected: len(mid) == 1, actual: 0
FAIL
PASS
PASS
--- FAIL: TestGetMidServers (0.00s)
    servers_test.go:475: getMidServers expected: len(mid) == 1, actual: 0
FAIL
PASS
PASS
PASS
PASS
--- FAIL: TestGetMidServers (0.00s)
    servers_test.go:475: getMidServers expected: len(mid) == 1, actual: 0
FAIL
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestGetMidServers (0.00s)
    servers_test.go:475: getMidServers expected: len(mid) == 1, actual: 0
FAIL
--- FAIL: TestGetMidServers (0.00s)
    servers_test.go:475: getMidServers expected: len(mid) == 1, actual: 0
FAIL
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestGetMidServers (0.00s)
    servers_test.go:475: getMidServers expected: len(mid) == 1, actual: 0
FAIL
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestGetMidServers (0.00s)
    servers_test.go:475: getMidServers expected: len(mid) == 1, actual: 0
FAIL
PASS
PASS
PASS
PASS
PASS
PASS
--- FAIL: TestGetMidServers (0.00s)
    servers_test.go:475: getMidServers expected: len(mid) == 1, actual: 0
FAIL
PASS
--- FAIL: TestGetMidServers (0.00s)
    servers_test.go:475: getMidServers expected: len(mid) == 1, actual: 0
FAIL
Test failed 10 times out of 100
```

<!--
Licensed to the Apache Software Foundation (ASF) under one
or more contributor license agreements.  See the NOTICE file
distributed with this work for additional information
regarding copyright ownership.  The ASF licenses this file
to you under the Apache License, Version 2.0 (the
"License"); you may not use this file except in compliance
with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing,
software distributed under the License is distributed on an
"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, either express or implied.  See the License for the
specific language governing permissions and limitations
under the License.
-->

--- a/traffic_ops/traffic_ops_golang/server/servers_test.go
+++ b/traffic_ops/traffic_ops_golang/server/servers_test.go
@@ -423,7 +423,13 @@ func TestGetMidServers(t *testing.T) {
 		serverMap[*server.ID] = server
 	}
 
-	ts := servers[1]
+	var ts tc.ServerNullable
+	for _, s := range servers {
+		if s.HostName != nil && *s.HostName == "server2" {
+			ts = s
+			break
+		}
+	}
 	*ts.ID = *ts.ID + 1
 	rows2 = rows2.AddRow(
 		ts.Cachegroup,

